src/backend/graphics/opengl.rs

```diff
@@ -0,0 +1,17 @@
+use nix::c_void;
+
+pub enum SwapBuffersError {
+    ContextLost,
+    AlreadySwapped,
+}
+
+pub trait OpenGLRenderer
+{
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError>;
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void;
+    fn get_framebuffer_dimensions(&self) -> (u32, u32);
+    fn is_current(&self) -> bool;
+    unsafe fn make_current(&self);
+    fn get_api(&self) -> Api;
+    fn get_pixel_format(&self) -> PixelFormat;
```

Comment:
I see you concerns. I didn't realise half of these methods where only required by glium.

I think the scope we should target for this backend is the union of the requirements of the various opengl libraries that may be used with smithay, *as long as it does not imply any significant overhead* (be it a runtime cost or a development cost).

But overall, I'd rather be in favour of providing too many methods than too few.